File: afcm-backend/app/schemas/nutrition.py
from typing import Literal

from pydantic import BaseModel, Field

from app.schemas.meal import DailyTotals, MealEntryOut

FitnessGoal = Literal["cut", "bulk", "maintain"]
HistoryRange = Literal["today", "yesterday", "last_7_days", "last_30_days"]


class CalorieTargets(BaseModel):
    cut: int
    maintain: int
    bulk: int


class MacroTargets(BaseModel):
    protein_g: int
    carbs_g: int
    fat_g: int


class NutritionPlan(BaseModel):
    goal: FitnessGoal
    daily_calories: int
    calorie_targets: CalorieTargets
    macro_targets: MacroTargets
    water_target_ml: int


class WaterSummary(BaseModel):
    date: str
    consumed_ml: int
    target_ml: int


class AddWaterRequest(BaseModel):
    amount_ml: int = Field(gt=0, le=5000)
    date: str | None = None


class DashboardData(BaseModel):
    date: str
    plan: NutritionPlan
    totals: DailyTotals
    water: WaterSummary
    entries: list[MealEntryOut]


class HistoryDay(BaseModel):
    date: str
    calories: int
    protein_g: float
    carbs_g: float
    fat_g: float
    water_ml: int
    weight_kg: float | None


class HistoryResponse(BaseModel):
    range: HistoryRange
    days: list[HistoryDay]
    calorie_target: int | None
    water_target_ml: int | None

File: frontend/src/types/nutrition.ts
import type { MealEntry } from './meal';
import type { FitnessGoal } from './user';

export interface CalorieTargets {
  cut: number;
  maintain: number;
  bulk: number;
}

export interface MacroTargets {
  protein_g: number;
  carbs_g: number;
  fat_g: number;
}

/** Returned by POST /nutrition/calculate (and embedded in dashboard/profile responses). */
export interface NutritionPlan {
  goal: FitnessGoal;
  /** Target for the selected goal */
  daily_calories: number;
  calorie_targets: CalorieTargets;
  macro_targets: MacroTargets;
  water_target_ml: number;
}

export interface DailyTotals {
  calories: number;
  protein_g: number;
  carbs_g: number;
  fat_g: number;
}

export interface WaterSummary {
  date: string;
  consumed_ml: number;
  target_ml: number;
}

export interface AddWaterRequest {
  amount_ml: number;
  date: string;
}

/** GET /dashboard */
export interface DashboardData {
  date: string;
  plan: NutritionPlan;
  totals: DailyTotals;
  water: WaterSummary;
  entries: MealEntry[];
}

export type HistoryRange = 'today' | 'yesterday' | 'last_7_days' | 'last_30_days';

export interface HistoryDay {
  date: string;
  calories: number;
  protein_g: number;
  carbs_g: number;
  fat_g: number;
  water_ml: number;
  /** Logged body weight for that day, if any */
  weight_kg: number | null;
}

/** GET /history?range= */
export interface HistoryResponse {
  range: HistoryRange;
  days: HistoryDay[];
  calorie_target: number | null;
  water_target_ml: number | null;
}
